ctx.go

```diff
@@ -378,7 +378,7 @@ func (c *Ctx) BodyParser(out interface{}) error {
 	ctype = utils.ParseVendorSpecificContentType(ctype)
 
 	// Parse body accordingly
-	if strings.HasPrefix(ctype, MIMEApplicationJSON) {
+	if strings.HasPrefix(ctype, MIMEApplicationJSON) || strings.HasPrefix(ctype, MIMEApplicationProblemJSON) {
```

Comment:
[RFC 6838](https://datatracker.ietf.org/doc/html/rfc6838#section-4.2.8) 
It might make sense to match anything that has the +json suffix here. e.g.
```go
var ctypeEnd int
if ctypeEnd = strings.IndexByte(ctype, ';'); ctypeEnd == -1 {
  ctypeEnd = len(ctype)
}
isJSON := strings.HasSuffix(utils.TrimRight(ctype, ' '), "+json")
```

Reply:
That works. I've rerun the test and updated the PR.